### ISSUE TITLE
Invalid frontmatter: `sidebar_label`

### ISSUE THREAD

**Issue by thomasheartman** [PaloAltoNetworks/docusaurus-openapi-docs#337]
## Describe the bug

When the OpenAPI summary prop is used to generate a sidebar label or title, it doesn't quote (or otherwise escape) the value, causing invalid values in some cases. 

Example (`title` is quoted and contains quoted values, which breaks the build):

```yaml
id: add-public-signup-token-user
title: "Create a user with the "viewer" root role and link them to a signup token"
description: "Create a user with the 'viewer' root role and link them to a signup token"
sidebar_label: Create a user with the "viewer" root role and link them to a signup token
hide_title: true
```

## Expected behavior

All frontmatter values that may cause YAML parsing to fail should be quoted and escaped. Generation shouldn't fail because of a character used in a summary.

## Current behavior

As demonstrated by this [failed Unleash documentation generation run](https://github.com/Unleash/unleash/actions/runs/3434792557/jobs/5726445104),  the YAML parser errors out with the following message:

```
Error:  Error while parsing Markdown front matter.
This can happen if you use special characters in front matter values (try using double quotes around that value).
Error:  Loading of version failed for version current
Error:  Unable to build website for locale en.
Error:  YAMLException: can not read a block mapping entry; a multiline key may not be an implicit key at line 4, column 12:
    description: "Create a user with the 'viewe ... 
```

Update: I am able to reproduce this locally. The error happens when trying to run `docusaurus build`.

For unquoted values, this same error occurs if you use a `:`.

## Possible solution

I think any text that can come from summaries, descriptions, or otherwise free text should be quoted when used as frontmatter. YAML always accepts text strings, so there is no downside to always quoting and escaping.

## Steps to reproduce

The generation fails on this endpoint (currently available at https://us.app.unleash-hosted.com/ushosted/docs/openapi.json, but won't necessarily be for long): 

```json
"/ushosted/invite/{token}/signup": {
      "post": {
        "tags": [
          "Public signup tokens"
        ],
        "operationId": "addPublicSignupTokenUser",
        "summary": "Create a user with the \"viewer\" root role and link them to a signup token",
        "requestBody": {
          "description": "createInvitedUserSchema",
          "required": true,
          "content": {
            "application/json": {
              "schema": {
                "$ref": "#/components/schemas/createInvitedUserSchema"
              }
            }
          }
        },
        "responses": {
          "200": {
            "description": "userSchema",
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/userSchema"
                }
              }
            }
          },
          "400": {
            "description": "The request data does not match what we expect."
          },
          "409": {
            "description": "The provided resource can not be created or updated because it would conflict with the current state of the resource or with an already existing resource, respectively."
          }
        },
        "parameters": [
          {
            "name": "token",
            "in": "path",
            "required": true,
            "schema": {
              "type": "string"
            }
          }
        ]
      }
    }
```

1. Use the endpoint description above (or a similar on with a summary containing quotes)
2. Generate docs using `docusaurus build`
3. See the error message you get.

## Context

This is currently [breaking Unleash's documentation build](https://github.com/Unleash/unleash/actions/runs/3434792557/jobs/5726445104)(and thus causing a number of failures on our main branch). Unless we remove the integration, we're currently unable to build new documentation, which is unfortunate.

For some reason, running `docusaurus start` does not cause any problems.

## Your Environment

- Version used: 1.4.1, 1.4.3, 0.0.0-506; docusaurus 2.1.0, 2.2.0
- Environment name and version (e.g. Chrome 59, node.js 5.4, python 3.7.3): Node 16.13.0
- Operating System and version (desktop or mobile): macOS
- Link to your project: https://github.com/Unleash/unleash

---

Please let me know if you need any more information ☺️ And as always: thanks a lot for your work on this plugin ❤️

**Comment by thomasheartman** [comment]
Just for the record, it seems the plugin will also happily create quoted strings with unescaped quotes inside. We had:

```
title: "A string title with "double quotes" inside"
```

This too breaks compilation as the yaml parser thinks that the value ends at the first inside quote.

**Comment by sserrata** [comment]
Thanks @thomasheartman, we had to address a similar/related issue with the `description` field. 

Basically, the solution is to always wrap in double quotes and regex replace any pre-existing double quotes in the value with single quotes. We'll see if we can work the fix into this next release train.

**Comment by thomasheartman** [comment]
Wonderful! And yeah, that's the workaround we ended up with as well. Thanks for your input; looking forward to seeing it land 😄

**Comment by sserrata** [comment]
Fix just released in `0.0.0-507`. Let me know if this addresses the issue.

**Comment by thomasheartman** [comment]
Yeah, nice! Just tried it out now and it works like a charm; thank you ❤️ 

And just a heads up: I did some more testing, and the `sidebar_label` wasn't actually what was breaking (YAML will happily accept quotes inside a string if it doesn't start with quotes, it seems), but the internal quotes in the already quoted `title`. However, the unquoted `sidebar_label` **would** break if you tried to insert a colon anywhere, so quoting it is absolutely the right way to go 😄
